src/commonMain/kotlin/fr/acinq/lightning/channel/InteractiveTx.kt

```diff
@@ -1075,7 +1068,17 @@ data class InteractiveTxSigningSession(
             val channelKeys = channelParams.localParams.channelKeys(keyManager)
             val unsignedTx = sharedTx.buildUnsignedTx()
             val sharedOutputIndex = unsignedTx.txOut.indexOfFirst { it.publicKeyScript == fundingParams.fundingPubkeyScript(channelKeys) }
-            val liquidityFees = liquidityLease?.fees?.total?.toMilliSatoshi() ?: 0.msat
+            val liquidityFees = liquidityLease?.let { l ->
+                val fees = l.fees.total.toMilliSatoshi()
+                when (l.paymentDetails) {
+                    is LiquidityAds.PaymentDetails.FromChannelBalance -> if (fundingParams.isInitiator) fees else -fees
+                    // Fees will be paid later, from relayed HTLCs.
+                    is LiquidityAds.PaymentDetails.FromFutureHtlc -> 0.msat
+                    is LiquidityAds.PaymentDetails.FromFutureHtlcWithPreimage -> 0.msat
+                    // Fees are taken from the current fee credit.
+                    is LiquidityAds.PaymentDetails.FromFeeCredit -> 0.msat
+                }
```

Comment:
This is a criticial part for the `FromChannelBalance` case, which counter-intuitively happens after the interactive-tx is built (but before it is signed). The reason being that fees are taken from the commitment tx, not from the funding tx.